### DISCUSSION TITLE
Do we make some changes to C# warning ca1063? with Stride.Core.DisposeBase

### DISCUSSION THREAD

**Discussion by Arc-huangjingtong** [stride3d/stride#2384] [General]
# C# warning CA1063
https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1063

# oringin code in Stride.Core.DisposeBase

``` csharp
    public void Dispose()
    {
        if (!IsDisposed)
        {
            this.ReleaseInternal();
        }
    }
```

# modified
``` csharp
    public void Dispose()
    {
        Dispose(true);
        GC.SuppressFinalize(this);
    }
    
    protected virtual void Dispose(bool disposing)
    {
        if (!IsDisposed)
        {
            this.ReleaseInternal();
        }
    }
```

Because some operations will prevent [System.IDisposable] from being called normally.

such as override Finalize ?

**Comment by Eideren** [comment]
We can also seal the class, which I think would make sense for the other instances of this warning we have in our codebase. No one's taking care of this as far as I know, feel free to tackle this one
